elk/training/train.py

```diff
@@ -78,88 +78,129 @@ def train_reporter(
+                )
+                pseudo_labels = torch.cat(
+                    [
+                        val_x0.new_zeros(n),
+                        val_x0.new_ones(n),
+                    ]
+                )
+                pseudo_labels = repeat(pseudo_labels, "n -> (n v)", v=v)
+                pseudo_auroc = float(
+                    roc_auc_score(pseudo_labels.cpu(), pseudo_preds.cpu())
+                )
+
+            if val_lm_preds is not None:
+                val_gt_cpu = (
+                    val_gt.repeat_interleave(val_lm_preds.shape[1]).float().cpu()
+                )
+                val_lm_auroc = float(roc_auc_score(val_gt_cpu, val_lm_preds.flatten()))
+                val_lm_acc = float(
+                    accuracy_score(val_gt_cpu, val_lm_preds.flatten() > 0.5)
+                )
+            else:
+                val_lm_auroc = None
+                val_lm_acc = None
+
+            row = pd.Series(
+                {
+                    "dataset": ds_name,
+                    "layer": layer,
+                    "pseudo_auroc": pseudo_auroc,
+                    "train_loss": train_loss,
+                    **val_result._asdict(),
+                    "lm_auroc": val_lm_auroc,
+                    "lm_acc": val_lm_acc,
+                }
+            )
 
-            lr_auroc, lr_acc = evaluate_baseline(lr_model, val_x0, val_x1, val_gt)
+            lr_auroc, lr_acc = evaluate_supervised(lr_model, val_x0, val_x1, val_gt)
 
             row["lr_auroc"] = lr_auroc
             row["lr_acc"] = lr_acc
-            save_baseline(lr_dir, layer, lr_model)
+            row_buf.append(row)
 
-        with open(reporter_dir / f"layer_{layer}.pt", "wb") as file:
-            torch.save(reporter, file)
+        return pd.DataFrame(row_buf)
 
-        return row
-
-    def get_pseudo_auroc(
-        self, layer: int, x0: Tensor, x1: Tensor, val_x0: Tensor, val_x1: Tensor
-    ):
+    def get_pseudo_classifier(self, data: dict[str, tuple], device: str) -> Classifier:
         """Check the separability of the pseudo-labels at a given layer."""
 
-        with torch.no_grad():
-            pseudo_auroc = Reporter.check_separability(
-                train_pair=(x0, x1), val_pair=(val_x0, val_x1)
-            )
-            if pseudo_auroc > 0.6:
-                warnings.warn(
-                    f"The pseudo-labels at layer {layer} are linearly separable with "
-                    f"an AUROC of {pseudo_auroc:.3f}. This may indicate that the "
-                    f"algorithm will not converge to a good solution."
-                )
+        x0s, x1s = [], []
+        for x0, x1, _, _ in data.values():
+            x0s.append(rearrange(x0, "n v d -> (n v) d"))
+            x1s.append(rearrange(x1, "n v d -> (n v) d"))
+
+        # Simple de-meaning normalization
+        X0 = torch.cat(x0s)
+        X1 = torch.cat(x1s)
+        X0 -= X0.mean(dim=0)
+        X1 -= X1.mean(dim=0)
```

Comment:
Yeahhhh but this is annoying because only CCS supports these different normalization options, I honestly kinda want to say that we should just require that CCS always use the normalization that they _said_ they used in the paper

Reply:
fixed